runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,420 @@
+				else if (client.getGameState().equals(GameState.LOGGED_IN) && emoteOnly && (mod || broadcaster) && isChatboxTransparent())
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=ffffff>[</col><col=9070ff>Twitch</col>] " + badges + config.channelName() + ": <col=" + Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+				}
+				//If chat is in emote only and the user is a mod or the broadcaster
+				else if (client.getGameState().equals(GameState.LOGGED_IN) && emoteOnly && (mod || broadcaster))
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=000000>[</col><col=0000FF>Twitch</col>] " + badges + config.channelName() + ": <col=" + Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+				}
+				//If the chat is in emote only mode on transparent chat box
+				else if (client.getGameState().equals(GameState.LOGGED_IN) && emoteOnly && isChatboxTransparent())
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=ffffff>[</col><col=9070ff>Twitch</col>] Notification: <col=3D2BAD>This channel is in emote only mode!<br><col=3D2BAD>Function coming soon to a Runelite client near you.</col>", "Twitch");
+				}
+				//If chat is in emote only mode
+				else if (client.getGameState().equals(GameState.LOGGED_IN) && emoteOnly)
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=000000>[</col><col=0000FF>Twitch</col>] Notification: <col=3D2BAD>This channel is in emote only mode!<br><col=3D2BAD>Function coming soon to a Runelite client near you.</col>", "Twitch");
+				}
+				//If chat is transparent and there are no room states
+				else if (client.getGameState().equals(GameState.LOGGED_IN) && isChatboxTransparent())
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=ffffff>[</col><col=9070ff>Twitch</col>] " + badges + config.channelName() + ": <col=" + Integer.toHexString(config.chatTColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+				}
+				//If chat is opaque and there are no room states
+				else if (client.getGameState().equals(GameState.LOGGED_IN))
+				{
+					client.addChatMessage(ChatMessageType.TRADE, "",
+							"<col=000000>[</col><col=0000FF>Twitch</col>] " + badges + config.channelName() + ": <col=" + Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+				}
+			}
+		}
+	}
+
+	//Shut off current running irc client
+	public void twitchShutdown(Client c)
+	{
+		try
+		{
+			c.shutdown();
+		}
+		catch (Exception e)
+		{
+			log.debug("Not connected to a chat yet.", e);
+		}
+	}
+
+	//Triggers when a message is sent to twitch chat
+	@Handler
+	public void onMessage(ChannelMessageEvent a)
+	{
+		String username = a.getActor().getUserString();
+
+		//Sub length badges
+		//0 months
+		if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/0"))
```

Comment:
Change `a.getOriginalMessages().get(0).getTag("badges").toString()` to a variable and use that for your if statements.